Program.cs

```diff
@@ -722,6 +722,97 @@ public static void Props()
+            temp = Console.ReadLine();
+
+            while (true)
+            {
+                Console.WriteLine("============================================");
+                Console.WriteLine("==            Entering the city           ==");
+                Console.WriteLine("==   1: Booking a room and have a rest    ==");
+                Console.WriteLine("==         2: Go and get HP energy        ==");
+                Console.WriteLine("==   3: Buy tools knifes, rods and fire   ==");
+                Console.WriteLine("============================================");
+                switch (temp)
+                {
+                    case "1":
+                        HP = HP + 40;
+                        break;
+
+                    case "2":
+                        HP = HP + 40;
+                        break;
+
+                    case "3":
+                        Console.WriteLine("tools packed");
+                        break;
+
+                    default:
+                        Console.WriteLine("==============================");
+                        Console.WriteLine("==  Continuing the journey  ==");
+                        Console.WriteLine("==============================");
+                        break;
+                }
+                Console.ReadLine();
+                Warehouse();
+            }
+        }//End of PowerUp
+
+        //Warehouse 
+        static void Warehouse()
+        {
+            Console.WriteLine("========================================");
+            Console.WriteLine("==  We on the halfway on the journey  ==");
+            Console.WriteLine("========================================");
+            bool answer = false;
+
+            while (answer == false)
+
+            {
+                Console.WriteLine("================================================================");
+                Console.WriteLine("==          Go on the map! and search if there's              ==");
+                Console.WriteLine("==           any warehouse near, we need to change            ==");
+                Console.WriteLine("==        some costume. Yes! Here is it. Go on 300 km         ==");
+                Console.WriteLine("==           Put it the GPS, Oh yeah! We get it here.         ==");
+                Console.WriteLine("==    Umm what a big warehouse. Surely got need new costume.  ==");
+                Console.WriteLine("==            Lets enter the door and see.                    ==");
+                Console.WriteLine("================================================================");
+                Console.WriteLine("==                 1: Asking for sales assistance.            ==");
+                Console.WriteLine("==                 2: Go on my own and try it.                ==");
+                Console.WriteLine("================================================================");
+            }
```

Comment:
While loop ends before the end of the menu options. This leads to an infinite loop of printing menus. Recommend moving the close bracket to before line 813